mo_net/train/backends/log.py

```
@@ -182,26 +217,65 @@ def log_iteration(
         )
-        self._session.commit()
+
+    def _log_iteration_sync(self, *, run_id: int, entry: LogEntry) -> None:
+        try:
+            with self._lock:
+                self._pending_entries.append(entry)
+                if len(self._pending_entries) >= self._batch_size:
+                    self._flush_batch(self._pending_entries, run_id=run_id)
+                    self._pending_entries.clear()
+        finally:
+            with suppress():
```

Ruff (0.11.9)

No arguments passed to `contextlib.suppress`. No exceptions will be suppressed and therefore this context manager is redundant

(B022)